core/src/main/java/org/hibernate/ogm/dialect/spi/BaseDialectNamingAdaptor.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.dialect.spi;
+
+public class BaseDialectNamingAdaptor implements DialectNamingAdaptor {
+
+	static final DialectNamingAdaptor DEFAULT_INSTANCE = new BaseDialectNamingAdaptor();
+
+	public BaseDialectNamingAdaptor() {
```

Comment:
Ah yes, sure. "protected" then maybe.

The public constructor together with the `DEFAULT_INSTANCE` constant made me stumble.